### ISSUE TITLE
chore: Address seaborn warnings

### ISSUE THREAD

**Issue by auguste-probabl** [probabl-ai/skore#2395]
The following example:
```python
from sklearn.datasets import load_iris
from sklearn.pipeline import Pipeline
from sklearn.preprocessing import PolynomialFeatures
from sklearn.linear_model import LogisticRegression
from skore import ComparisonReport, CrossValidationReport
X, y = load_iris(return_X_y=True)
estimator_1 = LogisticRegression(max_iter=10000,random_state=42)
estimator_2 = Pipeline(
    [
        ("poly", PolynomialFeatures()),
        ("predictor", LogisticRegression(max_iter=10000,random_state=0)),
    ]
)
report = ComparisonReport(
    [CrossValidationReport(estimator_1, X, y), CrossValidationReport(estimator_
2, X, y)]
)
display = report.inspection.coefficients()
display.plot()
```

yields:
```
/home/auguste/Desktop/work/skore/skore/.venv/lib/python3.13/site-packages/seaborn/axisgrid.py:855: FutureWarning:

Setting a gradient palette using color= is deprecated and will be removed in v0.14.0. Set `palette='dark:#1f77b4'` for the same effect.

  func(*plot_args, **plot_kwargs)
/home/auguste/Desktop/work/skore/skore/.venv/lib/python3.13/site-packages/seaborn/axisgrid.py:855: FutureWarning:

Setting a gradient palette using color= is deprecated and will be removed in v0.14.0. Set `palette='dark:#1f77b4'` for the same effect.

  func(*plot_args, **plot_kwargs)
```

It would be good to make this warning disappear; I think there are others that we can find by looking at the test suite warnings.

### See also

https://github.com/probabl-ai/skore/pull/2082

### Environment

main (fce4f566)

**Comment by Sharkyii** [comment]
@auguste-probabl can i proceed with this issue

**Comment by mohsinm-dev** [comment]
@auguste-probabl 

From what I see, the issue happens because `FacetGrid.map_dataframe()` automatically injects a `color` argument when calling the mapped function. When `sns.boxplot` receives both `color` and `hue`, seaborn 0.13 interprets this as a request to create a gradient palette from `color`. This pattern is deprecated in seaborn v0.14.

I think this can be resolved by explicitly setting `palette` in `boxplot_kwargs` when `hue` is not None (similar to how it is done for stripplot). In seaborn, `palette` takes priority over `color`, so this should prevent the gradient palette warning.

Visually, this should not change the output, since BOXPLOT_STYLE already overrides the box colors to black with transparent faces.

The remaining `PendingDeprecationWarning: vert: bool` appears to come from seaborn 0.13 internally passing `vert` to matplotlib 3.10. This seems to be a seaborn compatibility issue and likely needs to be addressed upstream.

**Comment by Sharkyii** [comment]
Thanks for the analysis @mohsinm-dev — I’ve implemented this approach to address the seaborn warning:

#2405 

Happy to adjust if you’d like a different direction or refinements!
